File: leetcode_150/array/remove_duplIcates_from_sorted_array_II_80.py
class Solution:
    def removeDuplicates(self, nums) -> int:
        k = 1
        c = 1
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                nums[k] = nums[i]
                c = 1
                k += 1

            elif nums[i] == nums[i-1]:
                if c < 2:
                    nums[k] = nums[i]
                    c += 1
                    k += 1
        print(k, nums)
            
ss = Solution()
ss.removeDuplicates([1,1,1,2,2,3])

File: leetcode_150/array/remove_duplIcates_from_sorted_array_II_80.java
class Solution {
    public int removeDuplicates(int[] nums) {
        var k = 1;
        var c = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                nums[k] = nums[i];
                c = 1;
                k++;
            } else if (nums[i] == nums[i - 1]) {
                if (c < 2) {
                    nums[k] = nums[i];
                    c++;
                    k++;
                }

            }
        }
        return k;
    }
}